src/timeMachine/selectors/index.test.ts

```diff
@@ -10,48 +14,178 @@ import {
+        now.getTime() + oneMillisecond
+      )
+    })
+
+    it('can parse a duration without now()', () => {
+      expect(handleCustomTime('-3d', now)).toEqual(now.getTime() - threeDays)
+      expect(handleCustomTime('+3d', now)).toEqual(now.getTime() + threeDays)
+
+      expect(handleCustomTime('-24h', now)).toEqual(
+        now.getTime() - twentyFourHours
+      )
+      expect(handleCustomTime('+24h', now)).toEqual(
+        now.getTime() + twentyFourHours
+      )
+
+      expect(handleCustomTime('-15m', now)).toEqual(
+        now.getTime() - fifteenMinutes
+      )
+      expect(handleCustomTime('+15m', now)).toEqual(
+        now.getTime() + fifteenMinutes
+      )
+
+      expect(handleCustomTime('-1ms', now)).toEqual(
+        now.getTime() - oneMillisecond
+      )
+      expect(handleCustomTime('+1ms', now)).toEqual(
+        now.getTime() + oneMillisecond
+      )
+    })
+
+    it('can ignore spaces around now and the sign correctly', () => {
+      expect(handleCustomTime(' now()', now)).toEqual(now.getTime())
+      expect(handleCustomTime('now() ', now)).toEqual(now.getTime())
+      expect(handleCustomTime('        now()  ', now)).toEqual(now.getTime())
+
+      expect(handleCustomTime(' now() -   3d', now)).toEqual(
+        now.getTime() - threeDays
+      )
+      expect(handleCustomTime('now()+   3d   ', now)).toEqual(
+        now.getTime() + threeDays
+      )
+
+      expect(handleCustomTime('-   24h', now)).toEqual(
+        now.getTime() - twentyFourHours
+      )
+      expect(handleCustomTime('           +  24h', now)).toEqual(
+        now.getTime() + twentyFourHours
+      )
+
+      expect(handleCustomTime(' 15m', now)).toEqual(
+        now.getTime() + fifteenMinutes
+      )
+      expect(handleCustomTime('15m          ', now)).toEqual(
+        now.getTime() + fifteenMinutes
+      )
+      expect(handleCustomTime('    15m          ', now)).toEqual(
+        now.getTime() + fifteenMinutes
+      )
+    })
+
+    it('can parse a JavaScript timestamp as a string', () => {
+      let specificTimeString: string = '2022-10-14T20:33:01.433Z'
+      expect(handleCustomTime(specificTimeString, now)).toEqual(
+        new Date(specificTimeString).getTime()
+      )
+
+      specificTimeString = '2022/07/04'
+      expect(handleCustomTime(specificTimeString, now)).toEqual(
+        new Date(specificTimeString).getTime()
+      )
+
+      specificTimeString = '2022-10-06 00:00'
+      expect(handleCustomTime(specificTimeString, now)).toEqual(
+        new Date(specificTimeString).getTime()
+      )
+    })
+
+    it('rejects numbers as invalid time', () => {
+      let numberString = '1666029415'
+      expect(() => handleCustomTime(numberString, now)).toThrowError()
+
+      numberString = '0'
+      expect(() => handleCustomTime(numberString, now)).toThrowError()
+
+      numberString = '-4'
+      expect(() => handleCustomTime(numberString, now)).toThrowError()
+    })
```

Comment:
To clarify: the util will throw an error. But the UI will not error out, because we have input validation elsewhere (which injects "ms" to any number).

We are doing this to explicitly reject Unix timestamps which are being deprecated in the Flux spec. Should this change, this test is a clue that we have to make changes in `handleCustomTime`.